ogr/ogrgeometry.cpp

```diff
@@ -4534,6 +4535,261 @@ OGRGeometryH OGR_G_Buffer(OGRGeometryH hTarget, double dfDist, int nQuadSegs)
+        {
+            int nStyle;
+            if (EQUAL(pszValue, "ROUND"))
+            {
+                nStyle = GEOSBUF_CAP_ROUND;
+            }
+            else if (EQUAL(pszValue, "FLAT"))
+            {
+                nStyle = GEOSBUF_CAP_FLAT;
+            }
+            else if (EQUAL(pszValue, "SQUARE"))
+            {
+                nStyle = GEOSBUF_CAP_SQUARE;
+            }
+            else
+            {
+                bParamsAreValid = false;
+                CPLError(CE_Failure, CPLE_NotSupported,
+                         "Invalid value for ENDCAP_STYLE: %s", pszValue);
+                break;
+            }
+
+            if (!GEOSBufferParams_setEndCapStyle_r(hGEOSCtxt, hParams, nStyle))
+            {
+                bParamsAreValid = false;
+            }
+        }
+        else if (EQUAL(pszParam, "JOIN_STYLE"))
+        {
+            int nStyle;
+            if (EQUAL(pszValue, "ROUND"))
+            {
+                nStyle = GEOSBUF_JOIN_ROUND;
+            }
+            else if (EQUAL(pszValue, "MITRE"))
+            {
+                nStyle = GEOSBUF_JOIN_MITRE;
+            }
+            else if (EQUAL(pszValue, "BEVEL"))
+            {
+                nStyle = GEOSBUF_JOIN_BEVEL;
+            }
+            else
+            {
+                bParamsAreValid = false;
+                CPLError(CE_Failure, CPLE_NotSupported,
+                         "Invalid value for JOIN_STYLE: %s", pszValue);
+                break;
+            }
+
+            if (!GEOSBufferParams_setJoinStyle_r(hGEOSCtxt, hParams, nStyle))
+            {
+                bParamsAreValid = false;
+                break;
+            }
+        }
+        else if (EQUAL(pszParam, "MITRE_LIMIT"))
+        {
+            try
+            {
+                std::size_t end;
+                double dfLimit = std::stod(pszValue, &end);
+
+                if (end != strlen(pszValue))
+                {
+                    throw std::invalid_argument("");
+                }
+
+                if (!GEOSBufferParams_setMitreLimit_r(hGEOSCtxt, hParams,
+                                                      dfLimit))
+                {
+                    bParamsAreValid = false;
+                    break;
+                }
+            }
+            catch (const std::invalid_argument &)
+            {
+                bParamsAreValid = false;
+                CPLError(CE_Failure, CPLE_IllegalArg,
+                         "Invalid value for MITRE_LIMIT: %s", pszValue);
+            }
+            catch (const std::out_of_range &)
+            {
+                bParamsAreValid = false;
+                CPLError(CE_Failure, CPLE_IllegalArg,
+                         "Invalid value for MITRE_LIMIT: %s", pszValue);
+            }
+        }
+        else if (EQUAL(pszParam, "QUADRANT_SEGMENTS"))
+        {
+            try
+            {
+                std::size_t end;
+                int nQuadSegs = std::stoi(pszValue, &end, 10);
```

Comment:
This parameter parsing/validation is really verbose and makes me wonder if I'm missing something like `std::optional<int> CPLReadInt(std::string_view)` or `bool CPLReadInt(const char*, int*)`.

`std::from_chars` is the best solution I've found but I recently ran into problems with it being unsupported on recent versions of Apple clang.

Reply:
> `std::from_chars` is the best solution I've found but I recently ran into problems with it being unsupported on recent versions of Apple clang.

well, it is now a requirement since it is used in apps/argparse/argparse.hpp

Otherwise, I wouldn't care much about validating that the provided string is in range and would just blindly use atoi() . We do that almost everywhere....